Lib/glyphsLib/builder/kerning.py

```diff
@@ -24,25 +24,34 @@ def to_ufo_kerning(self):
     for master in self.font.masters:
         master_id = master.id
         kerning_source = master.metricsSource.id  # Maybe be a linked master
-        if kerning_source in self.font.kerning:
-            kerning = self.font.kerning[kerning_source]
+        if kerning_source in self.font.kerningLTR:
+            kerning = self.font.kerningLTR[kerning_source]
             _to_ufo_kerning(self, self._sources[master_id].font, kerning)
+        if kerning_source in self.font.kerningRTL:
+            kerning = self.font.kerningRTL[kerning_source]
+            _to_ufo_kerning(self, self._sources[master_id].font, kerning, "RTL")
 
 
-def _to_ufo_kerning(self, ufo, kerning_data):
+def _to_ufo_kerning(self, ufo, kerning_data, direction="LTR"):
     """Add .glyphs kerning to an UFO."""
 
     warning_msg = "Non-existent glyph class %s found in kerning rules."
 
     for left, pairs in kerning_data.items():
-        match = re.match(r"@MMK_L_(.+)", left)
+        if direction == "LTR":
+            match = re.match(r"@MMK_L_(.+)", left)
+        elif direction == "RTL":
+            match = re.match(r"@MMK_R_(.+)", left)
```

Comment:
wait.. does Glyphs 3 really calls the class on the _left_-hand side `@MMK_R_` and the one on the _right_-hand side `@MMK_L_`?!

Reply:
I’ll try to answer: The kerning pairs are still defined in logical order. So the code logic doesn’t change here. First stays first, second stays second. Additionally, for the new RTL-specific kerning table, Georg has switched the `L` and `R` to now be visual rather than logical, (as he did with `kernLeft` and `kernRight` classes) which I believe shouldn't have happened. We should have silently accepted that `L` and `R` were never appropriate names but are used to describe 1st and 2nd.

FYI, you can check the two added `RTL_kerning_v2.glyphs` and `RTL_kerning_v3.glyphs` files for reference, in particular the kerning pair `reh-ar,hah-ar.init.swsh` which is defined in both.